Procyon.Reflection/src/main/java/com/strobel/reflection/emit/TypeBuilder.java

```diff
@@ -1231,14 +1232,10 @@ else if (!genericParameterBuilders.isEmpty()) {
 
             _hasBeenCreated = true;
 
-            _generatedClass = (Class<T>) getUnsafeInstance().defineClass(
-                fullName,
-                classBytes,
-                0,
-                classBytes.length,
-                Thread.currentThread().getContextClassLoader(),
-                _protectionDomain
-            );
+            MethodHandles.Lookup lookup = MethodHandles.lookup();
+            MethodHandles.Lookup privateLookup = MethodHandles.privateLookupIn(Class.forName(fullName), lookup);

```

Comment:
Did you test this?  I don't see how `Class.forName(fullName)` could work, since the class doesn't exist until the next line.

Reply:
hi!

it is definitely tested in usual workloads. Is there some cornercase you think is sure to hit those lines?

Reply:
@mstrobel : You have unit tests that cover that part of the code
@judovana: I'm confused how you can test this without changing JDK version from 1.7 to 11 in the build.gradle? I've run the build and I have 31 test failures (that can be reduced to 15 with the PR #35), failures such as variable renames that appear trivial at first but can cause collisions and loss of meaningful naming (x instead of b for boolean, etc...) and compiler methods such as makeConcatWithConstants not interpreted anymore